edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs

```diff
@@ -238,9 +249,57 @@ async void OnConnectionStatusChanged(ConnectionStatus status, ConnectionStatusCh
 
                 if (this.pullOnReconnect && this.initTask.IsCompleted && status == ConnectionStatus.Connected)
                 {
+                    var delayedTwinPull = true;
                     using (await this.twinLock.LockAsync())
                     {
-                        await this.RefreshTwinAsync();
+                        var now = DateTime.Now;
+                        if (now - this.lastTwinPullOnConnect > this.TwinPullOnConnectThrottleTime && !this.isDelayedTwinPullInProgress.Get())
+                        {
+                            this.lastTwinPullOnConnect = now;
+                            await this.RefreshTwinAsync();
+                            delayedTwinPull = false;
+                        }
+                    }
```

Comment:
nit: instead of this getting much longer as a method, could we split this into a helper method(s) like `HandleTwinAsync` or something? I am bad with names for things, but I think this could go in a helper for readability :)